modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,116 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import scala.annotation.tailrec
+import scala.util.control.NoStackTrace
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: Option[NonEmptyList[HistoricalSnapshotOrdinalEntry]])
+
+case class InvalidRange(msg: String) extends NoStackTrace {
+  override def getMessage: String = msg
+}
+case class InvalidRanges(msg: String) extends NoStackTrace {
+  override def getMessage: String = msg
+}
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      mkSnapshotOrdinalTimeline(s)
+        .leftMap(err => new DecoderError(err.getMessage))
+        .map(_.some)
+    }
+
+  @derive(eqv, show)
+  case class RangeBoundary(start: NonNegLong, end: NonNegLong) {
+    def overlaps(other: RangeBoundary): Boolean = other.start < end
```

Comment:
Why `<` rather than `<=` ? What is the expected result for `(a,b).overlaps(b,c)` ?

Reply:
`(a,b).overlaps(b,c)` is `true` if `b.end > b.start`. Basically, `x-100` does not overlap with `100-y`.

Reply:
> Basically, x-100 does not overlap with 100-y

Isn't it inclusive? Meaning `x-100` and `100-y` is `x-y`?

Reply:
Upon giving it more thought, you are correct. Changed to:

```scala
  def overlaps(other: Range): Boolean = other.start <= end
```